File: app/src/main/java/com/example/newsfeed/NewsApp/Database/Converter.java
package com.example.newsfeed.NewsApp.Database;

import androidx.room.TypeConverter;

import com.example.newsfeed.NewsApp.models.Source;

public class Converter {

    // passing source as a parameter and String is return-type
    @TypeConverter
    public String fromSource(Source source)
    {
return source.getName();
    }
    @TypeConverter
    public Source toSource(String name)
    {
        return new Source(name);
    }

}

File: app/src/main/java/com/example/newsfeed/NewsApp/Database/Converter.kt
package com.example.newsfeed.NewsApp.Database

import androidx.room.TypeConverter
import com.example.newsfeed.NewsApp.models.Source

class Converter {
    // passing source as a parameter and String is return-type
    @TypeConverter
    fun fromSource(source: Source): String {
        return source.name
    }
    @TypeConverter
    fun toSource(name: String): Source {
        return Source(name,name)
    }
}